cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,232 @@
+# Topic: CLI commands for User-Defined-Type definitions
+
+* **Author**: Reshma Abdul Rahim (reshrahim)
+
+## Topic Summary
+<!-- A paragraph or two to summarize the topic area. Just define it in summary form so we all know what it is. -->
+User-Defined Types enable Platform engineers to define and deploy their organization's custom services in Radius. More details around the feature, user scenarios and experience can be found [here](/architecture/2024-06-resource-extensibility-feature-spec.md). This document will focus on the CLI experience for creating and operating the User-Defined -Resource-Type definitions.
+
+### Top level goals
+<!-- At the most basic level, what are we trying to accomplish? -->
+- Provide an intuitive and seamless experience for platform engineers to create, update, and delete User-Defined-Type definitions using the CLI.
+- Ensure that the CLI experience is consistent with the other commands in Radius
+
+### Non-goals (out of scope)
+<!-- What are we explicitly not trying to accomplish? -->
+- Other interfaces and experiences to interact with user defined resource type definitions and resources are out of scope for this document
+
+## Key scenarios
+<!-- List ~3-7 high level scenarios to clarify the value and point to how we will decompose this big area into component capabilities. We may ultimately have more than one level of scenario. -->
+The key scenarios for UDTs are outlined [here](/architecture/2024-06-resource-extensibility-feature-spec.md). We will decompose these scenarios specific to the CLI experience.
+
+### Scenario 1: Deb defines and deploys the internal messaging service using the CLI
+<!-- One or two sentence summary -->
+Deb, a platform engineer who wants to define and deploy their internal messaging service Plaid as a resource type in Radius. 
+
+Before we start designing the experience, we need to understand the high level terminologies/concept that a user would interact with in the CLI experience.
+
+| Terminology | Description | Example | 
+|-------------|-------------|---------|
+|Resource provider | Refers to a group of resource types under a common namespace. Resource providers are also entities that implement the group of resource types | Mycompany.Messaging or Applications.Core |
+|Resource type | A service or technology that can be modeled as a resource | PostgreSQL or AWS S3 or Internal messaging service |
+|Resource | An instance of a resource type | mydb or mybucket |
+
+#### User Experience
```

Comment:
Refine the user exp to include the commands that a user would use 
1. rad resource-type create
2. call out support for both yaml and json